tech-far-hub/src/components/assessment.tsx

```diff
@@ -0,0 +1,276 @@
+    return errorMessage;
+  }
+
+  // Make sure we got a score page too
+  let assessmentQuestions: React.ReactElement;
+  let score: React.ReactElement;
+  if (typeof children[0] === "object" && "type" in children[0] && children[0].type === "ol") {
+    assessmentQuestions = children[0];
+  } else {
+    return errorMessage;
+  }
+  if (typeof children[1] === "object") {
+    score = children[1];
+  } else {
+    return errorMessage;
+  }
+
+  // Grab all the pages of the assessment (the assessment items)
+  let items = assessmentQuestions.props.children.filter(elementFilter);
+
+  if (
+    !items.every((item: React.ReactNode) => item && typeof item === "object" && "type" in item && item.type == "li")
+  ) {
+    return errorMessage;
+  }
+
+  // Each assessment page will need its visibility and contribution to the score tracked
+  const [isVisible, setIsVisible] = React.useState([true, ...Array(items.length + 1).fill(false)]);
+  const [scores, setScores]: [Number[], React.Dispatch<React.SetStateAction<Number[]>>] = React.useState(
+    Array(items.length).fill(1)
+  );
+
+  const handleNextPrevClick = (index: number): void => {
+    setIsVisible(() => {
+      const newIsVisible = Array(isVisible.length).fill(false);
+      newIsVisible[index] = true;
+      return newIsVisible;
+    });
+  };
+
+  const handleUpdateScore = (index: number, points: number): void => {
+    setScores(() => {
+      const newScores = [...scores];
+      newScores[index] = points;
+      return newScores;
+    });
+  };
+
+  // Pull the items into their own "screens" with the options
+  const screens = items.map((item: React.ReactNode, index: number) => {
+    let nextQuestionText = "Next Question";
+    let nextQuestionLink: React.ReactElement;
+    let atEnd = false;
+    if (index === items.length - 1) {
+      nextQuestionText = "Complete Assessment";
+      atEnd = true;
+    }
+    nextQuestionLink = (
+      <a
+        onClick={() => {
+          handleNextPrevClick(index + 1);
+        }}
+        className="tfh-assessment__next-question-link"
+      >
+        {nextQuestionText}
+      </a>
+    );
+    return (
+      <>
+        {isVisible[index] && (
+          <AssessmentItem itemNumber={index} handleUpdate={handleUpdateScore} currentScores={scores}>
+            {item}
+          </AssessmentItem>
+        )}
+        {isVisible[index] && nextQuestionLink}
+      </>
+    );
+  });
+
+  // Clone the rubric page so we can pass in the current score and a handler
+  const newScore = (
+    <AssessmentScore
+      scores={scores}
+      children={score.props.children}
+      clickHandler={handleNextPrevClick}
+    ></AssessmentScore>
+  );
+  // Add the rubric page to the end
+  screens.push(<>{isVisible[items.length] && newScore}</>);
+
+  return (
+    <>
+      <Card className="tfh-assessment">
+        <CardBody>{screens}</CardBody>
+      </Card>
+    </>
+  );
+};
+
+const AssessmentItem = ({
```

Comment:
This file looks good!

Wondering if it makes sense to move AssessmentItem and AssessmentScore into their own components? Not totally sure if it can work that way

Reply:
That is a good thought. I right now can't imagine them being used independently --- i mostly factored them out to keep the main Assessment component from being absurdly complex. I'll keep an eye out and do so if i see a chance to use them independently tho.